spec/index.bs

```diff
@@ -1326,6 +1331,77 @@ dictionary GPUBufferBinding {
 
   * {{GPUBufferBinding/size}}: If undefined, specifies the range starting at {{GPUBufferBinding/offset}} and ending at the end of the buffer.
 
+A {{GPUBindGroup}} object has the following internal slots:
+
+<dl dfn-type=attribute dfn-for="GPUBindGroup">
+    : <dfn>\[[layout]]</dfn> of type {{GPUBindGroupLayout}}>.
+    ::
+        The {{GPUBindGroupLayout}} this {{GPUBindGroup}} associates.
+
+    : <dfn>\[[bindings]]</dfn> of type sequence<{{GPUBindGroupBinding}}>.
+    ::
+        The set of {{GPUBindGroupBinding}}s this {{GPUBindGroup}} describes.
+</dl>
+
+### {{GPUDevice/createBindGroup()|GPUDevice.createBindGroup(GPUBindGroupDescriptor)}} ### {#GPUDevice-createBindGroup}
+
+<div algorithm="GPUDevice.createBindGroup(descriptor)">
+
+The <dfn method for="GPUDevice">createBindGroup(|descriptor|)</dfn> method is used to create {{GPUBindGroup}}s.
+
+1. Ensure [=bind group device validation=] is not violated.
+1. Let |bindGroup| be a new valid {{GPUBindGroup}} object.
+1. Ensure {{GPUBindGroupLayout}} in |descriptor|.{{GPUBindGroupDescriptor/layout}} is valid.
+1. Ensure the number of {{GPUBindGroupLayoutDescriptor/bindings}} of |descriptor|.{{GPUBindGroupDescriptor/layout}}
+    exactly equals to the number of |descriptor|.{{GPUBindGroupDescriptor/bindings}}.
+1. For each {{GPUBindGroupBinding}} |bindingDescriptor| in |descriptor|.{{GPUBindGroupDescriptor/bindings}}:
+    1. Ensure |bindingDescriptor|.{{GPUBindGroupBinding/binding}} does not violate [=bind group binding validation=].
+    1. If |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is {{GPUSampler}}:
+        1. Ensure it is a valid {{GPUSampler}} object.
+        1. Ensure [=sampler binding validation=] is not violated.
+    1. If |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is {{GPUTextureView}}:
+        1. Ensure it is a valid {{GPUTextureView}} object.
+        1. Ensure [=texture view binding validation=] is not violated.
+    1. If |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is {{GPUBufferBinding}}
+        1. Ensure it is a valid {{GPUBufferBinding}} object.
+        1. Ensure [=buffer binding validation=] is not violated.
+    1. Insert |bindingDescriptor| into |bindGroup|.{{GPUBindGroup/[[bindings]]}}.
+1. Return |bindGroup|.
+
+<b>Validation Conditions</b>
+
+<dl dfn-for="createBindGroup(descriptor)">
+If any of the following conditions are violated:
+    1. Generate a {{GPUValidationError}} in the current scope with appropriate error message.
+    1. Create a new invalid {{GPUBindGroup}} and return the result.
+
+<dfn>bind group device validation</dfn>: The {{GPUDevice}} must not be lost.
+
+<dfn>bind group binding validation</dfn>: Each |bindingDescriptor|.{{GPUBindGroupBinding/binding}} in |descriptor| must be unique.
```

Comment:
please wrap the lines to ~80 characters, so that we can review it on Github

Reply:
The spec doesn't obey this. I simply went through the spec, it turned out that some section wrapped the lines to 80 chars, and more sections wrapped to 100 chars. However, a lot more sections didn't wrap to a fixed number like 80 or 100 (but shorter than a much greater number like 300, :)). I wrap the <bind group creation> section to 100 chars, because 80 chars are not long enough sometimes.

Reply:
yeah, it's a weird restriction to place, but it's handy for our current workflow